src/context/apolloClient.ts

```diff
@@ -14,13 +14,35 @@ const cache = new InMemoryCache({
     Query: {
       fields: {
         getActionsByColony: {
-          keyArgs: false,
-          merge(existing = {}, incoming) {
+          keyArgs: ['$colonyAddress'],
+          /**
+           * The following function takes care of merging already fetched actions
+           * with the next set of paginated actions
+           */
+          merge(existing = {}, incoming, { readField, args }) {
```

Comment:
This is more or less following the docs: https://www.apollographql.com/docs/react/pagination/cursor-based#keeping-cursors-separate-from-items

In our case, the `getActionsByColony` doesn't return just an array but an object with a few fields. We store them in a map (id <> action), then transform them back before returning the results in the `read` function

Reply:
I like this, however I feel like the Cache file will keep growing, so I think it'll soon come time to separate it out in it's own module / multiple modules

Reply:
I extracted cache into its own file - as it grows, we can then separate it further into more logical chunks